chalk-solve/src/clauses/builtin_traits/sized.rs

```diff
@@ -14,22 +14,37 @@ fn push_adt_sized_conditions<I: Interner>(
 ) {
     let adt_datum = db.adt_datum(adt_id);
 
-    // ADTs with no fields are always Sized
-    if adt_datum.binders.skip_binders().fields.is_empty() {
+    // ADTs with no variants are always Sized
+    if adt_datum.binders.skip_binders().variants.is_empty() {
+        builder.push_fact(trait_ref.clone());
+        return;
+    }
+
+    // ADTs with only empty varaints are always Sized
+    if adt_datum
+        .binders
+        .skip_binders()
+        .variants
+        .iter()
+        .all(|variant| variant.fields.is_empty())
+    {
         builder.push_fact(trait_ref.clone());
         return;
     }
 
     let interner = db.interner();
 
-    // To check if an ADT type S<..> is Sized, we only have to look at its last field.
-    // This is because the WF checks for ADTs require that all the other fields must be Sized.
-    let last_field_ty = adt_datum
+    // To check if an ADT type S<..> is Sized, we only have to look at the last field of each
+    // of its variants. This is because the WF checks for ADTs require that all the other
+    // fields must be Sized.
+    let last_field_tys = adt_datum
```

Comment:
WF checks for enums require that *all* fields of all variants are `Sized`, so all well-formed enums are `Sized`. As a result, here we only need to check if the last field is `Sized` if the ADT is a struct.